ext/opentelemetry-ext-http-requests/src/opentelemetry/ext/http_requests/__init__.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+The opentelemetry-ext-requests package allows tracing HTTP requests made by the
+popular requests library.
+"""
+
+from urllib.parse import urlparse
+import functools
+
+from requests.sessions import Session
+
+
+# NOTE: Currently we force passing a tracer. But in turn, this forces the user
+# to configure a SDK before enabling this integration. In turn, this means that
+# if the SDK/tracer is already using `requests` they may, in theory, bypass our
+# instrumentation when using `import from`, etc. (currently we only instrument
+# a instance method so the probability for that is very low).
+def enable(tracer):
```

Comment:
could this be refactored to allow one to invoke injecting the tracer into a single requests Session?

I think the global patch should be a provided method as well, but I think it's important to allow consumers to configure multiple http clients. A global patch doesn't provide that flexibility.